forwarder/forwarder/queues/redis/tasks.py

```diff
@@ -0,0 +1,132 @@
+import json
+from enum import Enum
+
+from redis import StrictRedis
+
+
+class TaskState(Enum):
+    RECEIVED = "received"  # on receiving a task web-side
+    WAITING_FOR_EP = "waiting-for-ep"  # while waiting for ep to accept/be online
+    WAITING_FOR_NODES = "waiting-for-nodes"  # jobs are pending at the scheduler
+    WAITING_FOR_LAUNCH = "waiting-for-launch"
+    RUNNING = "running"
+    SUCCESS = "success"
+    FAILED = "failed"
+
+
+class Task:
+    """
+    ORM-esque class to wrap access to properties of tasks for better style and encapsulation
+
+    TODO: have a cache and TTL on the properties so that we aren't making so many redis gets?
+    """
+    def __init__(self, rc: StrictRedis, task_id: str, container: str = "", serializer: str = "", payload: str = ""):
+        """ If the kwargs are passed, then they will be overwritten.  Otherwise, they will gotten from existing
+        task entry.
+        Parameters
+        ----------
+        rc : StrictRedis
+            Redis client so that properties can get get/set
+        task_id : str
+            UUID of task
+        container : str
+            UUID of container in which to run
+        serializer : str
+        payload : str
+            serialized function + input data
+        """
+        self.rc = rc
+        self.task_id = task_id
+        self._task_hname = self._generate_hname(self.task_id)
+
+        # If passed, we assume they should be set (i.e. in cases of new tasks)
+        # if not passed,
+        if container:
+            self.container = container
```

Comment:
it seems `self.container` is not set if container is None, and this would break `self._generate_header`?

Reply:
Well yes and no; the intended usage was basically that if you don't pass those things, it means you don't know them and they will be fetched from Redis.  This would be the case when the user asks for a status knowing just the task id.  I think in all cases where we create tasks, we are explicitly setting container to something.  

I think now, the header is not actually used.  I thought it was bad style to place details of the task like that in what was effectively just the task id in the database.  So the new Tasks just use the task id, and store the container and serializer as keys under that task id